File: images/codex/acceptance/fixtures/python/tools/runtime_probe.py
from typing import Any, Literal

from pydantic import BaseModel, ConfigDict


description = "Report the Python authored-tool runtime."


class Input(BaseModel):
    model_config = ConfigDict(extra="forbid")


class Output(BaseModel):
    runtime: Literal["python"]


def execute(_: Input, _context: dict[str, Any]) -> Output:
    return Output(runtime="python")

File: images/codex/acceptance/fixtures/deno/tools/runtime_probe.ts
import { z } from "zod";

const inputSchema = z.object({}).strict();
const outputSchema = z.object({ runtime: z.literal("deno") }).strict();

export default {
  description: "Report the Deno authored-tool runtime.",
  inputSchema,
  outputSchema,
  execute() {
    return { runtime: "deno" as const };
  },
};
